File: 1/php/code.php
<?php

/**
* 
*/
function EulerNumber($integer)
{
	$current_total = 0;
	
	for ($i=0; $i < $integer; $i++) { 
		if (( $i % 3 == 0 ) || ( $i % 5 == 0 )){
			$current_total += $i;
		}
	}
	
	return $current_total;
}


?>

File: 1/ruby/code.rb
#!/usr/bin/env ruby
# 
# http://projecteuler.net/problem=1

class EulerNumber
  
  def initialize(integer)
    @number = integer
  end

  def sum_of_natural_multiples_of_three_or_five
    return (1...@number).to_a.keep_if {|v| (v % 3 == 0 || v % 5 == 0)}.reduce(:+)
  end
end